controllers/workspace/provision/deployment.go

```diff
@@ -167,6 +186,19 @@ func ScaleDeploymentToZero(workspace *dw.DevWorkspace, client runtimeClient.Clie
 	return nil
 }
 
+// SetDeploymentSpecStartedToFalse sets deployment.spec.started to false
+func SetDeploymentSpecStartedToFalse(workspace *dw.DevWorkspace, client runtimeClient.Client) error {
+	patch := []byte(`{"spec":{"started": false}}`)
+	err := client.Patch(context.Background(), &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: workspace.Namespace,
+			Name:      common.DeploymentName(workspace.Status.DevWorkspaceId),
+		},
+	}, runtimeClient.RawPatch(types.StrategicMergePatchType, patch))
+
+	return err
+}
```

Comment:
We want to apply this patch to the DevWorkspace, not the deployment -- I think this might be a no-op.

```suggestion
func SetDeploymentSpecStartedToFalse(workspace *dw.DevWorkspace, client runtimeClient.Client) error {
	patch := []byte(`{"spec":{"started": false}}`)
	err := client.Patch(context.Background(), workspace, runtimeClient.RawPatch(types.StrategicMergePatchType, patch))
	return err
}
```

This probably doesn't need a function of its own.

Reply:
> I think this might be a no-op.

WDYM? Currently, it's invoked only when workspace.Spec.Started.

Reply:
ohhhh okay done! i've removed the function as well!

Reply:
> WDYM? Currently, it's invoked only when workspace.Spec.Started.

I mean applying `{"spec":{"started": false}}` to a Deployment does nothing since the deployment doesn't have that field:
```
❯ kc patch deploy devworkspace-controller-manager --type merge -p '{"spec": {"started": false}}'                                             
deployment.apps/devworkspace-controller-manager patched (no change)
```

Reply:
It does not matter now but: as far as I saw workspace was patched but the function name was confused. So, it was not no-op.
But with removing this func with don't lose anything )